### PR TITLE
feat: add clientConfig to from verb

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
- Send at_client version to server

**- How I did it**
- Pass the client version via the clientConfig in AtLookupImpl in the RemoteSecondary.

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/util/at_client_util.dart [hunk 1]
```diff
@@ -135,6 +134,13 @@ class AtClientUtil {
     }
     return atKey.key!;
   }
+
+  static String getVersion() {
+    var applicationConfiguration = ApplicationConfiguration('pubspec.yaml');
```
Comment (gkc) [inline at_client/lib/src/util/at_client_util.dart]: This is dependent on what the "current working directory" is ...

Suggest you use this approach instead: https://stackoverflow.com/a/72486732

**Review by gkc — CHANGES_REQUESTED** [review]

File: at_client/lib/src/preference/at_client_config.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  static final AtClientConfig _singleton = AtClientConfig._internal();
+
+  AtClientConfig._internal();
+
+  factory AtClientConfig.getInstance() {
+    return _singleton;
+  }
+
+  /// Represents the at_client version.
+  final String atClientVersion = '3.0.36';
```
Comment (gkc) [inline at_client/lib/src/preference/at_client_config.dart:12]: Maintenance of this needs to be automated - maybe through using build_runner. At a minimum, you need to add a VERY VISIBLE comment to pubpsec.yaml stating that when the package version is increasing, then this constant needs to be changed accordingly.

**Review by gkc — CHANGES_REQUESTED** [review]

File: at_client/lib/src/util/at_client_util.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/util/at_client_util.dart]: Explored the above, the rootBundle is a flutter dependent package. Hence unable to use it in a Dart package.

File: at_client/lib/src/preference/at_client_config.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/preference/at_client_config.dart:12]: @gkc: For now, to get going, will add a comment in pubspec.yaml. And keep exploring the alternatives to automate the capturing the at_client version. Does this sound's good?

Comment (gkc) [reply at_client/lib/src/preference/at_client_config.dart:12]: 👍 thanks Sita

**Comment by sitaram-kalluri** [comment]
@gkc: Updated the pubspec.yaml with a comment. and changed the e2e test atSign from `ce2e4` to `ce2e2` because `ce2e4` runs on a prod version of at_secondary_server and hence the e2e tests are failing because the changes related to clientConfig are not yet released into Prod. I will a new PR having tests guarded with version checking to support cross version testing.

**Review by gkc — DISMISSED** [review]
OK for now

**Review by gkc — APPROVED** [review]

**Review by murali-shris — APPROVED** [review]

**Pull request merged**